drivers/serial/uart_stm32.c

```diff
@@ -1204,20 +1217,42 @@ static inline void async_timer_start(struct k_work_delayable *work,
 	}
 }
 
-static void uart_stm32_dma_rx_flush(const struct device *dev)
+static void uart_stm32_dma_rx_flush(const struct device *dev, int status)
 {
 	struct dma_status stat;
 	struct uart_stm32_data *data = dev->data;
 
-	if (dma_get_status(data->dma_rx.dma_dev,
-				data->dma_rx.dma_channel, &stat) == 0) {
-		size_t rx_rcv_len = data->dma_rx.buffer_length -
-					stat.pending_length;
+	size_t rx_rcv_len = 0;
+
+	switch (status) {
+	case DMA_STATUS_COMPLETE:
+		/* fully complete */
+		rx_rcv_len = data->dma_rx.buffer_length;
+		data->dma_rx.offset = 0;
+
+		break;
+	case DMA_STATUS_BLOCK:
+		/* half complete */
+		rx_rcv_len = data->dma_rx.buffer_length / 2;
+		data->dma_rx.offset = data->dma_rx.buffer_length / 2;
+
+		break;
+	default: /* likely ASYNC_UART_STATUS_TIMEOUT */
+		if (dma_get_status(data->dma_rx.dma_dev,
+				   data->dma_rx.dma_channel, &stat) == 0) {
+			rx_rcv_len = data->dma_rx.buffer_length -
+				     stat.pending_length;
+		}
```

Comment:
You're missing a `break;` here

Reply:
Default is the last case, nothing to do after. It is possible to use it here but I think it is not necessary.